feature_engine/variable_handling/_variable_type_checks.py

```diff
@@ -47,3 +48,11 @@ def _is_categorical_and_is_datetime(column: pd.Series) -> bool:
         is_dt = not _is_categories_num(column) and _is_convertible_to_dt(column)
 
     return is_dt
+
+
+def _raise_on_duplicated_column_names(X: pd.DataFrame, variables: Any) -> None:
+    df_check = X if variables is None else X[variables]
+    if isinstance(df_check, pd.Series):
+        return
+    if df_check.columns.duplicated().any():
+        raise ValueError("Input data contains duplicated variable names.")
```

Comment:
Quick questions:

Does df_check.columns.duplicated().any() raise an error on a series?

Is this implementation better practice? or would it be better to say:

```
    if not isinstance(df_check, pd.Series):
        if df_check.columns.duplicated().any():
            raise ValueError("Input data contains duplicated variable names.")
```

Reply:
a. It does raise an error, since `Series` objects do not have `columns`. Nevertheless, if slicing a `DataFrame` with a column name returns a `Series`, it means there was only one column with that name.
b. No idea. The linter didn't complain about the one in my initial commit.